game/definitions.rpy

```diff
@@ -741,45 +751,76 @@ init 0 python:
     def jn_is_time_block_early_morning():
         """
         Returns True if the current time is judged to be early morning.
+
+        Time blocks are absolute, and not modified by user preferences on sunrise/sunset.
         """
         return jn_get_current_hour() in range(3, 5)
 
     def jn_is_time_block_mid_morning():
         """
         Returns True if the current time is judged to be mid morning.
+
+        Time blocks are absolute, and not modified by user preferences on sunrise/sunset.
         """
         return jn_get_current_hour() in range(5, 9)
 
     def jn_is_time_block_late_morning():
         """
         Returns True if the current time is judged to be late morning.
+
+        Time blocks are absolute, and not modified by user preferences on sunrise/sunset.
         """
         return jn_get_current_hour() in range(9, 12)
 
     def jn_is_time_block_morning():
         """
         Returns True if the current time is judged to be morning generally, and not a specific time of morning.
+
+        Time blocks are absolute, and not modified by user preferences on sunrise/sunset.
         """
         return jn_get_current_hour() in range(3, 12)
 
     def jn_is_time_block_afternoon():
         """
         Returns True if the current time is judged to be afternoon.
+
+        Time blocks are absolute, and not modified by user preferences on sunrise/sunset.
         """
         return jn_get_current_hour() in range(12, 18)
 
     def jn_is_time_block_evening():
         """
         Returns True if the current time is judged to be evening.
+
+        Time blocks are absolute, and not modified by user preferences on sunrise/sunset.
         """
         return jn_get_current_hour() in range(18, 22)
 
     def jn_is_time_block_night():
         """
         Returns True if the current time is judged to be night.
+
+        Time blocks are absolute, and not modified by user preferences on sunrise/sunset.
         """
         return jn_get_current_hour() in range(22, 3)
 
+    def jn_is_day():
+        """
+        Returns True if the current time is judged to be day, taking into account user preferences on sunrise/sunset.
+        """
+        return datetime.time(persistent.jn_sunrise_hour) <= datetime.datetime.now().time() < datetime.time(persistent.jn_sunset_hour)
+
+    def jn_open_google_maps(latitude, longitude):
+        """
+        Opens Google Maps in a new tab/window in the default browser centred on the given latitude and longitude.
+        
+        IN:
+            - latitude - The latitude to centre the map on.
+            - longitude - The longitude to centre the map on.
+        """
+        url = "https://www.google.com/maps/place/{0},{1}".format(latitude, longitude)
+        webbrowser.open(url)
```

Comment:
Will need to test and see how this behaves on a platform-to-platform basis (so far win/linux look fine as a prelim test from their consoles)

Reply:
Understood